stackslib/src/net/api/getinfo.rs

```diff
@@ -217,16 +223,40 @@ impl RPCRequestHandler for RPCPeerInfoRequestHandler {
         node: &mut StacksNodeState,
     ) -> Result<(HttpResponsePreamble, HttpResponseContents), NetError> {
         let ibd = node.ibd;
-        let rpc_peer_info =
-            node.with_node_state(|network, _sortdb, chainstate, _mempool, rpc_args| {
-                RPCPeerInfoData::from_network(
+
+        let rpc_peer_info: Result<RPCPeerInfoData, StacksHttpResponse> =
+            node.with_node_state(|network, sortdb, chainstate, _mempool, rpc_args| {
+                let coinbase_height = NakamotoChainState::get_coinbase_height(
```

Comment:
Instead of performing an (expensive) MARF-indexed read on each request, can the node instead load and cache the coinbase height eagerly in `network.stacks_tip`, as part of the call to `PeerNetwork::refresh_burnchain_view()`?

Reply:
Addressed in [097c4db](https://github.com/stacks-network/stacks-core/pull/5308/commits/097c4db2e376f56794b0585347cb692585758bbe)